difffuzz/difffuzz_test.go

```diff
@@ -0,0 +1,41 @@
+package difffuzz
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/oasisprotocol/oasis-core/go/common/cbor"
+)
+
+func TestDecodingFromRust(t *testing.T) {
+	err := CborFromSlice([]byte{0x81, 0x18, 0x2a})
+	require.NoError(t, err)
+
+	err = CborFromSlice([]byte{0xde, 0xad, 0xbe, 0xef})
+	require.Error(t, err)
+}
+
+func FuzzDifferential(f *testing.F) {
+	// Seed corpus.
+	f.Add([]byte{0x81, 0x18, 0x2a})
+	f.Add([]byte{0x18, 0x2a})
+	f.Add([]byte{0xA1, 0x63, 0x66, 0x6F, 0x6F, 0x0A})
+	f.Add([]byte{0xA2, 0x65, 0x62, 0x79, 0x74, 0x65, 0x73, 0x41, 0x01, 0x63, 0x66, 0x6F, 0x6F, 0x18, 0x2A})
+
+	// Fuzzing.
+	f.Fuzz(func(t *testing.T, data []byte) {
+		var output map[interface{}]interface{}
+		err := cbor.Unmarshal(data, &output)
+		if err != nil {
+			return
+		}
+
+		// If decoding succeeded, make sure it also succeeds in the Rust version.
```

Comment:
Should it also hold true that if decoding fails in go, it should also fail in Rust, or no?

Reply:
This is less of a problem because the main reason for this whole consistency thing is that the consensus layer logic uses the Go deserializer when parsing and then storing that value into consensus state. So if parsing in Go fails, it won't even come to the Rust side.

The problem with testing the fail/fail consistency is currently that this crate's non-strict version will allow many more things to be decoded because the Go side is representing CBOR maps using Go maps which have restrictions on what types can be used as keys (when map types are left as `interface{}`). In practice this is not a problem because we always define the types on the Go side but we can't do this during fuzzing easily.

Reply:
Makes sense, thanks for explaining!